src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -396,6 +406,30 @@ The **`stage`** denotes how this script (the `content`) should be inserted. Some
 
 A read-only copy of the user-supplied [Astro config](/en/reference/configuration-reference/). This is resolved _after_ other integrations have run.
 
+#### `setAdapter` option
+
+**Type:** `(adapter: AstroAdapter) => void;`
+
+Makes the integration an adapter. Read more in the [adapter API](/en/reference/adapter-reference/).
+
+#### `injectTypes` options
+
+<p><Since v="4.14.0" /></p>
+
+**Type:** `(injectedType: { filename: string; content: string }) => URL`
+
+Allows you to inject types easily in your user's project. `filename` must ends with `".d.ts"` and `content` must be valid TypeScript (it will be formatted).
+
+Under the hood, it will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.
```

Comment:
```suggestion
 This will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.
```
I would put this BELOW the code example, since we've just described what the `filename` and `content` should look like in the example above. Probably a good time to let them see it in action right here, THEN explain what this resulting code will do.

Reply:
There was so much to explain in there, I was not sure how to organize it! Whatever you prefer